### PR TITLE
feat(endpoints): initial endpoints

### PR DESCRIPTION
**Pull request by drisc**
Also updates the readme to be python specific with basic information

### CODE REVIEW

**Comment by wescopeland** [comment]
May want to look into something like autopep8 to enforce PEP 8 compliance.

File: retroachievements/client.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
+import requests as request
+
+
+class RAClient:
+
+    # Header (required to pass cloudflare consistantly)
```
Comment (luchaos) [inline retroachievements/client.py]: Technically this is not related to Cloudflare

File: retroachievements/client.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    headers = {
+        "User-Agent": "api-python/0.1 - RetroAchievements Web API Python Wrapper"
```
Comment (luchaos) [inline retroachievements/client.py]: "User-Agent": "retroachievements-api-python/1.0 - RetroAchievements Web API Python Client"

@wescopeland we should probably standardize this across clients as well?

File: retroachievements/client.py
Comment (wescopeland) [inline retroachievements/client.py:1]: `params={}` is mutable by default, which could be error-prone. For example:

```python
def append_to_list(value, my_list=[]):
    my_list.append(value)
    return my_list

# First call with default list
print(append_to_list(1))  # Expected: [1], Output: [1]

# Second call with default list
print(append_to_list(2))  # Expected: [2], Output: [1, 2]
```

A default value of `None` is safer for dictionaries and lists, eg:

```python
def append_to_list(value, my_list=None):
    if my_list is None:
        my_list = []
    my_list.append(value)
    return my_list
```

File: retroachievements/client.py (hunk 2 above)
Comment (wescopeland) [reply retroachievements/client.py]: We should indeed. I propose:

```
RetroAchievements-api-js/1.X.X (any other useful metadata here)
```

If this is agreeable, I will get it into api-js right away.

Comment (luchaos) [reply retroachievements/client.py]: Sounds good!

I'd aim for a 1.0 release here right away as well. No need to hold back with major versions imo.

Comment (wescopeland) [reply retroachievements/client.py]: 👍 Tracking in api-js at https://github.com/RetroAchievements/api-js/issues/75

Comment (drisc) [reply retroachievements/client.py]: String updated with suggestion

File: retroachievements/client.py (hunk 1 above)
Comment (drisc) [reply retroachievements/client.py]: Cloudflare comment removed

File: retroachievements/__init__.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+RetroAchievements API Wrapper
+---
+
+Official Python API wrapper for RetroAchievements
+
+"""
+
+__title__ = "retroachievements-api"
+__authors__ = "drisc"
+__version__ = "0.1"
```
Comment (wescopeland) [inline retroachievements/__init__.py]: semver may be preferable.

Comment (luchaos) [reply retroachievements/__init__.py]: Can this be referenced in the user agent directly? I assume this makes it easier to maintain the version number.

Comment (drisc) [reply retroachievements/__init__.py]: Was just about to commit that update haha

File: retroachievements/__init__.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
+"""
+
+RetroAchievements API Wrapper
+---
+
+Official Python API wrapper for RetroAchievements
```
Comment (luchaos) [inline retroachievements/__init__.py]: "wrapper" or "client"?

Comment (wescopeland) [reply retroachievements/__init__.py]: Good catch. The docs will always refer to this as a client library.

Comment (drisc) [reply retroachievements/__init__.py]: At this point I'd call it a wrapper, it's just spitting out the HTTP response in a format that Python applications can make use of. If you want I can change it to "client"

Comment (luchaos) [reply retroachievements/__init__.py]: Personally, I prefer the term "client" - not a strong opinion though. In this case it would be fitting for consistency's sake.

Comment (wescopeland) [reply retroachievements/__init__.py]: Let's go with client -- it's plausible this package could also do some sane type-setting kind of like api-js does, for example:

```
// HTTP API
"Untracked": 0
```

```
// api-js
"untracked": false
```

Comment (drisc) [reply retroachievements/__init__.py]: Description updated

File: retroachievements/__init__.py (hunk 3 above)
Comment (drisc) [reply retroachievements/__init__.py]: The user-agent now references the __version__ in __init__.py

Comment (wescopeland) [reply retroachievements/__init__.py]: That is a good change. It's a separate issue from the original comment, though. It may be preferable for this package to use semantic versioning.

**Comment by drisc** [comment]
> May want to look into something like autopep8 to enforce PEP 8 compliance.

I've installed autopep8 in my code editor, is there a way to add it as a push step for this repo similar to the test suite that RAWeb has?

**Comment by wescopeland** [comment]
> I've installed autopep8 in my code editor, is there a way to add it as a push step for this repo similar to the test suite that RAWeb has?

Yes, that sounds like a natural next step. Unfortunately, my python CI knowledge is not very good.

Two things will likely be required:
* A pre-commit hook (which may be doable via the `pre-commit` pip package, but I am not entirely sure.
* A GitHub CI actions pipeline to verify PEP8 compliance.

**Comment by luchaos** [comment]
I can add one of those to see if they yield the expected results?
https://github.com/marketplace/actions/pep8-action
https://github.com/marketplace/actions/pycodestyle

File: retroachievements/client.py
Comment (drisc) [reply retroachievements/client.py:1]: I've implemented the fix suggested here, url_params are no longer mutable by default

File: retroachievements/__init__.py (hunk 3 above)
Comment (drisc) [reply retroachievements/__init__.py]: I updated it to use semver in the same commit: https://github.com/RetroAchievements/api-python/pull/1/files#diff-40d00a57b07939167fe7ec4058a6d4cee495453a268e1c9bd7123857b4b92525R12

Comment (wescopeland) [reply retroachievements/__init__.py]: Ah, my mistake. I did not see it. Thank you 🙏

**Review by wescopeland — APPROVED** [review]
